packages/core/src/new-api/types/journal.ts

```diff
@@ -49,6 +52,8 @@ export type OnchainInteractionMessage =
 export interface DeployContractInteractionMessage {
   type: "onchain-action";
   subtype: "deploy-contract";
+  futureId: string;
+  transactionId: number;
```

Comment:
What is this `transactionId`? If this is meant to be like an internal nonce, I really think we should come up with another name that doesn't conflict with existing Ethereum concepts.

Reply:
Agreed it is a bad name.

It is trying to give an id to a desired on-chain interaction for a future. So the first message yielded in the execution strategy would be 1, and the second 2. This lets us know when we are rerunning the same on-chain interaction.

Reply:
Maybe `OnchainId`? Do you have another suggestion?

Reply:
`executionId` or `messageId` maybe?

Reply:
Alright, I will go with `executionId`, I will push it in a later PR to reduce rebase pain.